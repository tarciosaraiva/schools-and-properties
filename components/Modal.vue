<template>
  <transition name="modal">
    <div class="modal-mask">
      <div class="modal-wrapper">
        <div class="modal-container">
          <div class="modal-header">
            <slot name="header"> default header </slot>
          </div>

          <div class="modal-body">
            <slot name="body"> default body </slot>
          </div>

          <div class="modal-footer text-right">
            <slot name="footer">
              <button class="pl-button full small" @click="$emit('close')">
                Close
              </button>
            </slot>
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>

<script lang="ts">
import Vue from 'vue'
export default Vue.extend({
  mounted() {
    document.addEventListener('keyup', this.handleEscape)
  },
  destroyed() {
    document.removeEventListener('keyup', this.handleEscape)
  },
  methods: {
    handleEscape(event: any) {
      if (event.keyCode === 27) {
        this.$emit('close')
      }
    },
  },
})
</script>
