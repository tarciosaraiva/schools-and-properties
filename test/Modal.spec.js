import { mount } from '@vue/test-utils'
import Modal from '~/components/Modal.vue'

// const localVue = createLocalVue()

describe('Modal', () => {
  let wrapper
  const closeFn = jest.fn()

  beforeEach(() => {
    const ModalTestComponent = {
      template: '<modal @close="close"></modal>',
      components: { Modal },
      methods: { close: closeFn }
    }

    wrapper = mount(ModalTestComponent)
  })


  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('emits "close" event when clicking on close button', () => {
    const button = wrapper.find('button')
    button.trigger('click')
    expect(closeFn).toHaveBeenCalled()
  })

  test('emits "close" event when sending escape key', () => {
    wrapper.trigger('keyup.esc')
    expect(closeFn).toHaveBeenCalled()
  })
})
