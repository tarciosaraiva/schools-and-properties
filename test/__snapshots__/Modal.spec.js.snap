// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should match snapshot 1`] = `
<transition-stub name="modal">
  <div class="modal-mask">
    <div class="modal-wrapper">
      <div class="modal-container">
        <div class="modal-header"> default header </div>
        <div class="modal-body"> default body </div>
        <div class="modal-footer text-right">
          <button class="pl-button full small">
            Close
          </button>
        </div>
      </div>
    </div>
  </div>
</transition-stub>
`;
